crates/hir-ty/src/method_resolution.rs

```diff
@@ -1058,6 +1058,8 @@ fn iterate_method_candidates_by_receiver(
     mut callback: &mut dyn FnMut(ReceiverAdjustments, AssocItemId, bool) -> ControlFlow<()>,
 ) -> ControlFlow<()> {
     let receiver_ty = table.instantiate_canonical(receiver_ty);
+    let is_receiver_ty_unknown = TyFingerprint::for_trait_impl(&receiver_ty).is_none();
```

Comment:
this seems a really roundabout way to check for unknown? if we mean unbound type variables, just `receiver_ty.inference_var(Interner).is_some()` should do it after the instantiate_canonical, or `receiver_ty.is_general_var(Interner, &receiver_ty.binders)` before it. ... but actually, if the receiver_ty was unknown, we wouldn't reach this point in the first place because of the early return in `iterate_method_candidates_with_autoref`, would we? I'm not sure I understand the logic here. (`TyFingerprint::for_trait_impl(&receiver_ty).is_none()` just includes a bunch of other type kinds that probably shouldn't be?)

Reply:
Unfortunately, using either of your suggestions here causes the following test to fail:

```
---- syntax_highlighting::tests::benchmark_syntax_highlighting_parser stdout ----
syntax highlighting parser: 4.64s, 0b
thread 'syntax_highlighting::tests::benchmark_syntax_highlighting_parser' panicked at crates/ide/src/syntax_highlighting/tests.rs:1231:5:
assertion `left == right` failed
  left: 1167
 right: 1169
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    syntax_highlighting::tests::benchmark_syntax_highlighting_parser
```

In fact, I only introduced this code involving `is_receiver_ty_unknown` because, having made my other changes, I found that this exact test was failing. I did some investigation involving comparing the test data generated by this test before and after applying my other changes. I found that `TyFingerprint::for_trait_impl(&receiver_ty).is_none()` was sufficient to ensure this test passed.

Reply:
That test failing is generally fine I believe. The assertion there is not really meaningful for the most part in terms of correctness as there are a bunch of unresolved types in the fixture of it anyways. I am actually not even sure why we have the assert in there in the first place, as its meant as a benchmark.

Reply:
Ok sure, I will remove this code involving `is_receiver_ty_unknown` entirely. I will instead update the assertion count in the test as you suggest.